frontend/src/views/WheresMyGeneV2/components/HeatMap/index.tsx

```diff
@@ -301,87 +295,58 @@ export default memo(function HeatMap({
     return [...result].sort().map((cellType) => ({ name: cellType }));
   }, [sortedCellTypesByTissueName]);
 
+  /**
+   * All tissue cell types to render in YAxisCharts
+   */
+  const allTissueCellTypes = useMemo(() => {
+    return displayedTissues
+      .sort((a, b) => {
+        // sort tissues alphabetically
+        return a.name.localeCompare(b.name);
+      })
+      .map((tissue: OntologyTerm) => {
+        const { id, name } = tissue;
+
+        return {
+          tissueId: id,
+          tissueName: name,
+          tissueCellTypes: getTissueCellTypes({
+            cellTypeSortBy,
+            cellTypes,
+            sortedCellTypesByTissueName,
+            tissue: name,
+            tissueID: id,
+            displayedCellTypes,
+          }),
+        };
+      });
+  }, [
+    cellTypeSortBy,
+    cellTypes,
+    sortedCellTypesByTissueName,
+    displayedCellTypes,
+    displayedTissues,
+  ]);
+
   const handleFilteredCellTypesChange = (
     _: unknown,
     rawNewFilteredCellTypes: DefaultAutocompleteOption[]
   ) => {
-    if (!dispatch) return;
-
     const cellTypeNames = rawNewFilteredCellTypes.map(
       (cellType) => cellType.name
     );
     const cellTypeIds = cellTypeNames.map((name) => cellTypesByName[name].id);
 
-    dispatch(
+    dispatch?.(
       setFilteredCellTypes({
         filteredCellTypes: cellTypeNames,
         filteredCellTypeIds: cellTypeIds,
+        displayedTissueIds,
       })
     );
   };
-  useEffect(() => {
-    if (filteredCellTypes.length === 0) {
-      setDisplayedCellTypes(initialDisplayedCellTypeIds);
-      setExpandedTissues(EMPTY_SET as Set<string>);
-    }
-  }, [
-    filteredCellTypes.length,
-    initialDisplayedCellTypeIds,
-    setExpandedTissues,
-  ]);
-
-  // Reset `displayedCellTypes` and `expandedTissues` when the user clears `filteredCellTypes`
-  useEffect(() => {
```

Comment:
`filteredCellTypes` reducer will update state `expandedTisssueIds` when `filteredCellTypes` changes, which will then trigger a recalculation of `displayedCellTypes` through `useMemo`. This way we have a single source of truth for `displayedCellTypes` calculation